syntax = "proto3";

option java_multiple_files = true;
option java_package = "";
option java_outer_classname = "";
option objc_class_prefix = "";

package package;

service RecognizerServer {

    rpc Recognize (RecognitionData) returns (RecognizedFaces) {}

}

message RecognitionData {
    string clientId = 1;
    int32 imageId = 2;
    bytes bytes = 3;
}

enum RecognizerRC {
    OK = 0;
    UNAUTHORIZED = 1;
    FINISHED = 2;
}

message RecognizedFaces {
    RecognizerRC rc = 1;
    repeated Face faces = 2;
    uint32 total = 3;
    uint32 unique = 4;
}

message Face {
    string name = 1;
    double confidence = 2;
    int32 x = 3;
    int32 y = 4;
    int32 width = 5;
    int32 height = 6;
}
